From: Ales Novak <alnovak@suse.cz>
Date: Tue, 18 Mar 2014 15:42:47 +0100
Patch-mainline: Never, upstream has more advanced, locking-less function
Subject: Unlock the rename_lock in dentry_path() in the case when path is too long
References: bnc#868748

This bug is fixed in upstream by commit 232d2d60aa5469bb097f55728f65146bd49c1d25

Signed-off-by: Ales Novak <alnovak@suse.cz>
Acked-by: Miklos Szeredi <mszeredi@suse.cz>

---
 fs/dcache.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/dcache.c b/fs/dcache.c
index 9082273..9dc5ac3 100644
--- a/fs/dcache.c
+++ b/fs/dcache.c
@@ -2863,6 +2863,7 @@ char *dentry_path(struct dentry *dentry, char *buf, int buflen)
 		*p = '/';	/* restore '/' overriden with '\0' */
 	return retval;
 Elong:
+	write_sequnlock(&rename_lock);
 	return ERR_PTR(-ENAMETOOLONG);
 }
 
